src/Handlebars.Net.Helpers/HandlebarsHelpers.cs

```diff
@@ -180,5 +182,20 @@ private static object InvokeMethod(HandlebarsHelpersOptions options, string name
                 throw;
             }
         }
+
+        /// <summary>
+        /// For Handlebars.Net : also return a helperName surrounded by [ ] to support the official handlebarsjs rules
+        /// For Handlebars.CSharp : only register normal helper name
+        ///
+        /// See https://github.com/StefH/Handlebars.Net.Helpers/issues/7
+        /// </summary>
+        private static string[] CreateHelperNames(string helperName)
+        {
+#if HANDLEBARSNET
+            return new[] { helperName, $"[{helperName}]" };
```

Comment:
You do not need to register helper wrapped with [ ] - only helper ref in the template itself should be wrapped

Reply:
This is needed for the original Handlebars.net because that one does not understand `[String.Append]`.
(When I don't do this, unit-tests will fail for Handlebars.net)

Reply:
Strange, I definitely know this functionality is included into the latest revision. Probably it’s yet included into the package.